concent_api/core/payments/bankster.py

```diff
@@ -264,80 +235,115 @@ def settle_overdue_acceptances(
+        sum_of_existing_requestor_claims = DepositClaim.objects.filter(
+            payer_deposit_account=requestor_deposit_account
+        ).aggregate(
+            sum_of_existing_claims=Coalesce(Sum('amount'), 0)
+        )
 
-    current_time = get_current_utc_timestamp()
+        # If the existing claims against requestor's deposit are greater or equal to his current deposit,
+        # we can't add a new claim.
+        if requestor_deposit_value <= sum_of_existing_requestor_claims['sum_of_existing_claims']:
+            return None
 
-    # Concent gets list of transactions from payment API where timestamp >= T0.
-    list_of_transactions = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
-        requestor_eth_address=requestor_ethereum_address,
-        provider_eth_address=provider_ethereum_address,
-        payment_ts=oldest_payments_ts,
-        current_time=current_time,
-        transaction_type=TransactionType.BATCH,
-    )
+        # Concent defines time T0 equal to oldest payment_ts from passed SubtaskResultAccepted messages from
+        # subtask_results_accepted_list.
+        oldest_payments_ts = min(subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances)
 
-    # Concent defines time T1 equal to youngest timestamp from list of transactions.
-    if len(list_of_transactions) == 0:
-        t1_is_bigger_than_payments_ts = False
-    else:
-        youngest_transaction = max(transaction.closure_time for transaction in list_of_transactions)
-        # Concent checks if all passed SubtaskResultAccepted messages from subtask_results_accepted_list
-        # have payment_ts < T1
-        t1_is_bigger_than_payments_ts = any(
-            youngest_transaction > subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances
+        current_time = get_current_utc_timestamp()
+
+        # Concent gets list of transactions from payment API where timestamp >= T0.
+        list_of_transactions = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
+            requestor_eth_address=requestor_ethereum_address,
+            provider_eth_address=provider_ethereum_address,
+            payment_ts=oldest_payments_ts,
+            current_time=current_time,
+            transaction_type=TransactionType.BATCH,
         )
 
-    # Any of the items from list of overdue acceptances
-    # matches condition current_time < payment_ts + PAYMENT_DUE_TIME
-    acceptance_time_overdue = any(
-        current_time < subtask_results_accepted.payment_ts + settings.PAYMENT_DUE_TIME
-        for subtask_results_accepted in acceptances
-    )
+        # Concent defines time T1 equal to youngest timestamp from list of transactions.
+        if len(list_of_transactions) == 0:
+            t1_is_bigger_than_payments_ts = False
+        else:
+            youngest_transaction = max(transaction.closure_time for transaction in list_of_transactions)
```

Comment:
please rename variable `transaction`. There is imported `transaction` from `django.db` and my IDE shows warning, that this variable shadows imported one. I know it works, but I think it should be renamed. BTW. Does your IDE shows warning in this place as well?

Reply:
:+1: yes, it was showing it too.